pygments/lexers/markup.py

```diff
@@ -760,3 +763,69 @@ def _handle_cssblock(self, match):
     def __init__(self, **options):
         self.handlecodeblocks = get_bool_opt(options, 'handlecodeblocks', True)
         RegexLexer.__init__(self, **options)
+
+class TexinfoLexer(RegexLexer):
+    """
+    For `Texinfo <https://gnu.org/software/texinfo>`_ source code.
+
+    .. versionadded:: 2.10
+    """
+
+    name = 'Texinfo'
+    aliases = ['texi', 'texinfo']
+    filenames = ['*.texi']
+    mimetypes = ['text/x-texi']
+
+    tokens = {
+        'base': [
+            (r'\s+', Text),
+            (r'@$', Keyword),
+            (r'@[@:{}*]', Keyword),
+            (r'(?s)``.*?''', String),
+        ],
+        'commands': [
+            (r'(@[a-z]+)(\{)',
+             bygroups(Keyword, Punctuation), 'command'),
+            (r'@[a-z]+', Keyword)
+        ],
+        'root': [
+            (r'@c\s.*?$', Comment),
+            (r'@ignore.*?@end ignore', Comment),
+            include('base'),
+            (r'(@)(node|section|subsection|subsubsection|part|chapter|centerchap|unnumbered|unnumberedsec|unnumberedsubsec|unnumberedsubsection|unnumberedsubsubsec|unnumberedsubsubsection|chapheading|majorheading|heading|subheading|subsubheading|appendix|appendixsec|appendixsection|appendixsubsec|appendixsubsection|appendixsubsubsec|appendixsubsubsection|[cfvkpt]index|printindex|syncodeindex)(\s)',
```

Comment:
There is a `words()` function for matching lists like this. See https://pygments.org/docs/lexerdevelopment/#handling-lists-of-keywords.